presto-druid/src/main/java/com/facebook/presto/druid/DruidSplitManager.java

```diff
@@ -52,16 +69,119 @@ public ConnectorSplitSource getSplits(
     {
         DruidTableLayoutHandle layoutHandle = (DruidTableLayoutHandle) layout;
         DruidTableHandle table = layoutHandle.getTable();
-        if (isComputePushdownEnabled(session) || (table.getDql().isPresent() && table.getDql().get().getPushdown())) {
+        if (isComputePushdownEnabled(session) && (table.getDql().isPresent() && table.getDql().get().getPushdown())) {
             return new FixedSplitSource(ImmutableList.of(createBrokerSplit(table.getDql().get())));
         }
-        List<String> segmentIds = druidClient.getDataSegmentId(table.getTableName());
 
+        final List<String> segmentIds;
+        DruidTableLayoutHandle tableLayout = (DruidTableLayoutHandle) layout;
+        Map<String, List<Object>> dimFilters = new HashMap<>();
+        if (tableLayout.getTupleDomain().getDomains().isPresent()) {
+            String segmentFilter = null;
+            for (Map.Entry<ColumnHandle, Domain> entry : tableLayout.getTupleDomain().getDomains().get().entrySet()) {
+                DruidColumnHandle handle = (DruidColumnHandle) entry.getKey();
+                if (handle.getColumnName().equals("__time")) {
+                    Domain predicate = entry.getValue();
+                    segmentFilter = segmentFilter(predicate);
+                }
+                else {
+                    DruidColumnType dct = DruidColumnType.fromPrestoType(handle.getColumnType());
+                    if (dct == DruidColumnType.VARCHAR) {
```

Comment:
According to the spec, numbers are as either dimensions or metrics are stored as arrays. So to filter off the array, I'd still have to traverse every entry to determine whether to return segment to presto. However, since the filtering is at every split then we'd divide the work amongst the worker nodes and would see substantial gain IF presto filters after aggregating all the sliced rows into one. I guess my question is does presto filter at every slice or  does it merge all the slices into one and then begins the predicate evaluation. (I haven't check the code in this aspect so I'll take a look).